File: security-in-action/auth-stream/demo/src/main/java/com/example/demo/domain/Member.java
package com.example.demo.domain;

import com.example.demo.domain.enums.AlgorithmType;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.HashSet;
import java.util.Set;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class Member {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String username;
    private String password;

    @Enumerated(EnumType.STRING)
    private AlgorithmType algorithm;

    @OneToMany(mappedBy = "member", fetch = FetchType.EAGER)
    private Set<Authority> authorities = new HashSet<>();

    @Builder
    public Member(String username, String password, AlgorithmType algorithm, Set<Authority> authorities) {
        this.username = username;
        this.password = password;
        this.algorithm = algorithm;
        this.authorities = authorities;
    }
}

File: spring-boot-jpa2-restapi/restapi/src/main/kotlin/com/example/restapi/domain/Member.kt
package com.example.restapi.domain

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.Column
import jakarta.persistence.Embedded
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany
import jakarta.validation.constraints.NotEmpty

@Entity
class Member(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    val id: Long? = null,

    @Column(name = "member_name")
    @NotEmpty
    var name: String? = null,

    @Embedded
    var address: Address? = null,

    // @JsonIgnore -> 엔티티에 이런 로직이 들어가는 것은 좋지 않다
    @OneToMany(mappedBy = "member")
    var orders: List<Order> = arrayListOf()
)